src/hooks/useEnvironmentToggle.tsx

```diff
@@ -0,0 +1,32 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { useEffect, useState } from 'react';
+import { NativeEventEmitter } from 'react-native';
+
+export const useEnvironmentToggle = () => {
+  const [usePrimaryEnv, setUsePrimaryEnv] = useState(true);
+
+  useEffect(() => {
+    const eventEmitter = new NativeEventEmitter({
+      addListener: () => {},
+      removeListeners: () => {},
+    });
```

Comment:
Should this happen outside of the hook and have it be exported so that `ToggleEnvironmentMenuItem.tsx` can use that reference instead of making its own new instance?